.github/workflows/main.yml

```diff
@@ -0,0 +1,147 @@
+name: Build BLE examples
+
+on:
+  pull_request:
+  push:
+
+jobs: 
+  prepare:
+    container:
+      image: ghcr.io/armmbed/mbed-os-env:master-latest
+
+    outputs:
+      matrix : ${{ steps.find-examples.outputs.matrix }}
+
+    runs-on: ubuntu-latest
+
+    steps:
+      -
+        name: Checkout mbed-os-ble-utils
+        uses: actions/checkout@v2
+        with:
+          path: mbed-os-ble-utils
+
+      - name: Checkout mbed-os
+        uses: actions/checkout@v2
+        with:
+          repository: ARMmbed/mbed-os
+          path: mbed-os
+
+      - name: Checkout mbed-os-example-ble
+        uses: actions/checkout@v2
+        with:
+          repository: ARMmbed/mbed-os-example-ble
+          path: mbed-os-example-ble
+
+      - name: set examples in json object
+        id: find-examples
+        shell: bash
+        run: |
+          set -x
+          cd mbed-os-example-ble
+          examples=""
+          for i in BLE_* ; do
+            examples="{\"name\":\"$i\"},$examples"
+          done
+          examples="[${examples::-1}]"
+          echo "::set-output name=matrix::$examples"
+
+      - name: Prepare repositories
+        run: |
+          set -x
+          cd mbed-os-example-ble
+          # move .libs to temporary
+          # symlink mbed-os and mbed-os-ble-utils
+          # do mbed deploy
+          # move temporary to .libs
+          for i in BLE_* ; do
+            cd $i ;
+            mv mbed-os.lib mbed-os.lib.tmp
+            if [ -f mbed-os-ble-utils.lib ]; then
+              mv mbed-os-ble-utils.lib mbed-os-ble-utils.lib.tmp
+            fi
+            ln -s ../../mbed-os mbed-os ;
+            ln -s ../../mbed-os-ble-utils mbed-os-ble-utils ;
+
+            mbed config root . ;
+            mbed deploy
+            mv mbed-os.lib.tmp mbed-os.lib
+            if [ -f mbed-os-ble-utils.lib.tmp ]; then
+              mv mbed-os-ble-utils.lib.tmp mbed-os-ble-utils.lib
+            fi            
+            cd .. ;
+          done
+
+      - name: compress repos
+        run: |
+          tar -czf repos.tgz mbed-os-ble-utils mbed-os mbed-os-example-ble
```

Comment:
I think it would be possible to use cache. But that will do it for now. It is a small overhead compared to compilation.